nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/InvokeHTTP.java

```diff
@@ -788,6 +801,9 @@ public void onTrigger(ProcessContext context, ProcessSession session) throws Pro
             }
         }
 
+        if (context.getProperty(PROP_READ_TIMEOUT).isExpressionLanguagePresent())
+            timeout = context.getProperty(PROP_READ_TIMEOUT).evaluateAttributeExpressions(requestFlowFile).asTimePeriod(TimeUnit.MILLISECONDS).intValue();
```

Comment:
@guvencenanguvenal As far as I can tell the static shared timeout is going to create a concurrency issue. I think we should be using `newBuilder()` here instead of an interceptor (see [here](https://square.github.io/okhttp/3.x/okhttp/okhttp3/OkHttpClient.html) and [here](https://github.com/square/okhttp/blob/b3dcb9b1871325248fba917458658628c44ce8a3/docs/recipes.md#per-call-configuration-kt-java)). For example:
```suggestion
        if (context.getProperty(PROP_READ_TIMEOUT).isExpressionLanguagePresent()) {
            final int connectTimeout = context.getProperty(PROP_CONNECT_TIMEOUT).evaluateAttributeExpressions(requestFlowFile).asTimePeriod(TimeUnit.MILLISECONDS).intValue();        
            final int readTimeout = context.getProperty(PROP_READ_TIMEOUT).evaluateAttributeExpressions(requestFlowFile).asTimePeriod(TimeUnit.MILLISECONDS).intValue();         
            okHttpClient = okHttpClient.newBuilder()
                    .connectTimeout(connectTimeout, TimeUnit.MILLISECONDS)            
                    .readTimeout(readTimeout, TimeUnit.MILLISECONDS)
                    .build();
        }
```